tests/system/feature_store/test_spark_engine.py

```diff
@@ -332,6 +333,69 @@ def test_basic_remote_spark_ingest(self):
         print(f"expected_stats_df: {expected_stats_df.to_json()}")
         assert stats_df.equals(expected_stats_df)
 
+    def test_parquet_filters(self):
+        parquet_source_path = self.get_pq_source_path()
+        df = pd.read_parquet(parquet_source_path)
+        filtered_df = df.query('department == "01e9fe31-76de-45f0-9aed-0f94cc97bca0"')
+
+        base_path = self.get_test_output_subdir_path()
+        parquet_target_path = f"{base_path}_spark"
+        parquet_source = ParquetSource(
+            "parquet_source",
+            path=parquet_source_path,
+            additional_filters=[
+                ("department", "=", "01e9fe31-76de-45f0-9aed-0f94cc97bca0")
+            ],
+        )
+        feature_set = fstore.FeatureSet(
+            "parquet-filters-fs", entities=[fstore.Entity("patient_id")], engine="spark"
+        )
+
+        target = ParquetTarget(
+            name="department_based_target",
+            path=parquet_target_path,
+            partitioned=True,
+            partition_cols=["department"],
+        )
+        run_config = fstore.RunConfig(local=self.run_local)
+        feature_set.ingest(
+            source=parquet_source,
+            targets=[target],
+            spark_context=self.spark_service,
+            run_config=run_config,
+        )
+        result = mlrun.datastore.utils.sort_df(
+            pd.read_parquet(feature_set.get_target_path()), "patient_id"
+        )
+        expected = mlrun.datastore.utils.sort_df(filtered_df, "patient_id")
+        assert_frame_equal(result, expected, check_dtype=False, check_categorical=False)
+
+        vec = fstore.FeatureVector(
+            name="test-fs-vec", features=["parquet-filters-fs.*"]
+        )
+        vec.save()
+        get_offline_target = ParquetTarget(
+            "mytarget", path=f"{self.output_dir()}-get_offline_features"
+        )
+        kind = None if self.run_local else "remote-spark"
+        resp = fstore.get_offline_features(
+            feature_vector=vec,
+            additional_filters=[("bad", "=", 95)],
```

Comment:
I guess better test a more complex one, to really test the filter logic. I'm thinking at least two filters, some thing with `not in` and something with NaN (which so much of the logic is dedicated to).

Reply:
Still missing a NaN value in the test data afaict. Not to be confused with `None`.